homeassistant/components/yolink/binary_sensor.py

```diff
@@ -153,4 +146,17 @@ def update_entity_state(self, state: dict[str, Any]) -> None:
         self._attr_is_on = self.entity_description.value(
             state.get(self.entity_description.state_key)
         )
+        # check door sensor paired device
+        if (
+            self.coordinator.device.parent_id is not None
+            and self.coordinator.device.parent_id != "null"
+            and self.coordinator.device.device_type == ATTR_DEVICE_DOOR_SENSOR
+        ):
+            paired_device_coordinator = self.hass.data[DOMAIN][
+                self.config_entry.entry_id
+            ][ATTR_COORDINATORS].get(self.coordinator.device.parent_id)
+            if paired_device_coordinator is not None:
+                paired_device_coordinator.async_set_updated_data(
```

Comment:
It's weird that we update the coordinator state from this entity. I'd expect that to happen from a central place in the integration, not from a specific entity.

Data should normally flow from the coordinator to the entities and not the other way around.

What's the goal here?

Reply:
this is because garage door controller device will not save it's state, it's state dependency on paired device such as door sensor or contact sensor.

Reply:
Doesn't the library know that they are paired? The platforms in Home Assistant aren't allowed to depend on each other. I would expect the library to handle this already if this relationship is meant.

Reply:
Yes, I have finish yolink api update , here are yolink api code change.

https://github.com/YoSmart-Inc/yolink-api/compare/v0.1.5...v0.2.1.b1